src/runtime_src/core/common/api/xrt_module.cpp

```diff
@@ -706,12 +728,17 @@ class module_elf : public module_impl
   }
 
   bool
-  patch(uint8_t* base, const std::string& argnm, uint64_t patch, patcher::buf_type type) override
+  patch(uint8_t* base, const std::string& argnm, size_t index, uint64_t patch, patcher::buf_type type) override
   {
     const std::string key_string = generate_key_string(argnm, type);
     auto it = m_arg2patcher.find(key_string);
-    if (it == m_arg2patcher.end())
-      return false;
+    auto found_use_argument_name = (it == m_arg2patcher.end());
+    if (false == found_use_argument_name) {// Search using index
```

Comment:
this should be 
true ==

After change this, the test passed

Reply:
Huh :-)  No, it should be `if (cond)`